BrowserKit/Sources/WebEngine/WKWebview/WKEngineSession.swift

```diff
@@ -371,6 +376,7 @@ class WKEngineSession: NSObject,
     func webView(_ webView: WKWebView,
```

Comment:
Seems great! I would add a follow up task to add unit test for this use case when we have https://mozilla-hub.atlassian.net/browse/FXIOS-8477 as well

Reply:
Good point, I added a task; also added an inline TODO just to make sure this stays on the radar